src/worker/worker.ts

```diff
@@ -84,21 +75,40 @@ export class Worker {
   private calculateOverviews(bbox: BBox2d, zoomLevel: number): number[] {
     const lonDiff = bbox[2] - bbox[0];
     const latDiff = bbox[3] - bbox[1];
-
     let overviewFactor = 2;
     const overviews: number[] = [];
 
     // Overviews are built 1 zoom level before the maximum zoom level
     const maxOverviewZoom = zoomLevel - 1;
     let pixelSize = getPixelResolution(maxOverviewZoom);
+    let overviewCounter = maxOverviewZoom;
 
-    while (lonDiff >= pixelSize && latDiff >= pixelSize) {
+    while (lonDiff >= pixelSize && latDiff >= pixelSize && overviewCounter >= 0) {
       overviews.push(overviewFactor);
       overviewFactor = overviewFactor << 1;
       // eslint-disable-next-line @typescript-eslint/no-magic-numbers
       pixelSize = pixelSize * 2;
+      overviewCounter--;
     }
 
     return overviews;
   }
+
+  private async *tilesGenerator(rangeGen: Iterable<ITileRange>, tilesDirectory: string): AsyncGenerator<Tile> {
+    for (const range of rangeGen) {
+      const tilesInZoomLevel = tilesCountPerZoom(range.zoom);
+      for (let x = range.minX; x <= range.maxX; x++) {
```

Comment:
it should be < and not <=

Reply:
the loops may be replaced by mc-utils tilesGenerator and single loop

Reply:
Cannot be replaced since I want to create a generator that already has BLOB in it, using mc-utils generator made it over complicated and unreadable

Reply:
the loops can still be replaced not the entire function